source/containers.tex

```diff
@@ -9200,6 +9238,665 @@
+\remarks
+If an exception is thrown other than by the
+copy constructor,
+move constructor,
+assignment operator, or
+move assignment operator
+of \tcode{T} or by
+any \tcode{InputIterator} operation,
+there are no effects.
+Otherwise,
+if an exception is thrown, then
+$\tcode{size()} \ge n$ and
+%FIXME: Is "begin() + [0, n)" a valid range?
+elements in the range \tcode{begin() + \range{0}{$n$}} are not modified.
+\end{itemdescr}
+
+\indexlibrarymember{push_back}{inplace_vector}%
+\indexlibrarymember{emplace_back}{inplace_vector}%
+\begin{itemdecl}
+constexpr reference push_back(const T& x);
+constexpr reference push_back(T&& x);
+template<class... Args>
+  constexpr reference emplace_back(Args&&... args);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\returns
+\tcode{back()}.
+
+\pnum
+\throws
+\tcode{bad_alloc} or
+any exception thrown by initialization of inserted element.
+
+\pnum
+\complexity
+Constant.
+
+\pnum
+\remarks
+If an exception is thrown there are no effects on \tcode{*this}.
+\end{itemdescr}
+
+\indexlibrarymember{try_emplace_back}{inplace_vector}%
+\indexlibrarymember{try_push_back}{inplace_vector}%
+\begin{itemdecl}
+template<class... Args>
+  constexpr pointer try_emplace_back(Args&&... args);
+constexpr pointer try_push_back(const T& x);
+constexpr pointer try_push_back(T&& x);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+Let \tcode{vals} denote a pack:
+\begin{itemize}
+\item \tcode{std::forward<Args>(args)...} for the first overload,
+\item \tcode{x} for the second overload,
+\item \tcode{set::move(x)} for the third overload.
+\end{itemize}
+
+\pnum
+\expects
+\tcode{T} is \oldconcept{EmplaceConstructible}
+into \tcode{*this} from \tcode{vals...}.
+
+\pnum
+\effects
+If \tcode{size() < capacity()} is \tcode{true},
+appends an object of type \tcode{T}
+direct-non-list-initialized with \tcode{vals...}.
+Otherwise, there are no effects.
+
+\pnum
+\returns
+\keyword{nullptr} if \tcode{size() == capacity()} is \tcode{true},
+otherwise \tcode{addressof(back())}.
+
+\pnum
+\throws
+Nothing unless an exception is thrown by initialization of inserted element.
+
+\pnum
+\complexity
+Constant.
+
+\pnum
+\remarks
+If an exception is thrown there are no effects on \tcode{*this}.
+\end{itemdescr}
+
+\indexlibrarymember{try_append_range}{inplace_vector}%
+\begin{itemdecl}
+template<@\exposconcept{container-compatible-range}@<T> R>
+  constexpr ranges::borrowed_iterator_t<R> try_append_range(R&& rg);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\expects
+\tcode{T} is \oldconcept{EmplaceConstructible}
+from \tcode{*ranges::begin(rg)} into \tcode{*this}.
+
+\pnum
+\effects
+Appends copies of initial elements in \tcode{rg} before \tcode{end()},
+until all elements are inserted or \tcode{size() == capacity()} is \tcode{true}.
+Each iterator in the range \tcode{rg} is dereferenced at most once.
+
+\pnum
+\returns
+%FIXME: Should this be "An iterator past the last inserted element of rg."?
+Iterator past last inserted element of \tcode{rg}.
```

Comment:
```suggestion
An iterator that points one past the copy of the last element inserted into `*this`, or `end()` if `rg` is empty.
```

Cf. the wording for [`a.insert(p, i, j)`](https://eel.is/c++draft/sequence.reqmts#38) and [`a.insert_range(p, rg)`](https://eel.is/c++draft/sequence.reqmts#42).

Reply:
Here, below (line 9774), and also in the Complexity element (line 9764), I wonder if we should say "the _successfully_ inserted element[s]" instead of just "the inserted element[s]."

Reply:
please forward to LWG.